components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -4069,15 +4085,17 @@ private OAuth2Parameters getOAuth2ParamsFromOAuthMessage(OAuthMessage oAuthMessa
      * @param params       OAuth2 Parameters
      * @return PKCE code challenge. Priority will be given to the value inside the OAuth2Parameters.
      */
-    private String getPkceCodeChallenge(OAuthMessage oAuthMessage, OAuth2Parameters params) {
+    private String getPkceCodeChallenge(OAuthMessage oAuthMessage, OAuth2Parameters params)
+            throws InvalidRequestException {
 
-        String pkceChallengeCode;
+        String pkceChallengeCode = null;
+        boolean isFapiConformantApp = isFapiConformant(params.getClientId());
         // If the code_challenge is in the request object, then it is added to Oauth2 params before this point.
         if (params.getPkceCodeChallenge() != null) {
             // If Oauth2 params contains code_challenge get value from Oauth2 params.
             pkceChallengeCode = params.getPkceCodeChallenge();
-        } else {
-            // Else retrieve from request query params.
+        } else if (!isFapiConformantApp) {
```

Comment:
Do the validation only if pkce mandated

Reply:
fixed with ef0e046274f92b950041433c237956dc64e81944